### ISSUE TITLE
DK Zone Regression + Remy gossip text

### ISSUE THREAD

**Issue by zimms** [unified-db/Database#72]
Using latest ACID, UDB, Core, and SD2.

DK zone regression:
Quests after forging the runeblade seem broken.  Specifically, unworthy initiates do not respond at all after unlocking their shackles.  Thought this was a one-off problem, so used some DB editing to "complete" this quest.  The next quest, the eye of acherus, doesn't work at all - the eye doesn't go anywhere, and also has the wrong model.  I stopped there.  Did not have this issue before DK update 11 when I last updated.
Wondering if the recent "delete duplicate" commits could be the cause?  Also wondering if anyone else can repro (takes literally 2 mins).

Remy "Two Times" gossip text:
Somehow he still has the same gossip text as Marshal Dughan.

**Comment by Grz3s** [comment]
Hi @zimms 
Thanks for your report.
1st.
Just tested 'The endless Hunger' - the one with unworthy initiates -- and works fine..no errors at all
2nd : Eye of Acherus: - same as above. - works fine (only speed for eye doesnt work - but this is core problem)
also wrong model for eye was fixed here: 
https://github.com/scriptdev2/scriptdev2/commit/b788229974caddb8e589ec4533f8655e9554eb61
ANyway .. both scripts are made by ScriptDev2 scripts.

All DK Zone Updates up to 12 ... they're made for DB .. they do not change anything in sd2 scripts, so if something worked for you before... theres no way that any UDB update damage them.

3rd:
Remy's gossip has been fixed here:
https://github.com/unified-db/Database/commit/c62c88725444c07b1612ac7754eb5bb45161bdb4

What i can suggest you to to:
1.Compile fresh cmangos core + fresh sd2 -- then use "InstallFullDB.sh"(very simple tool) to bring your DB and ACID up to date... and then check all again.
2. show us your "Server.log" -- after starting your server. 
Pls use http://paste2.org/
Maybe there we'll find what is wrong with your sd2.

**Comment by zimms** [comment]
Thanks for the prompt reply Grz3s!  I rebuilt everything and reimported the latest SD2 SQL + UDB + ACID.  The DK quests work now, I think something was corrupted or out of sync with the last update I did - sorry for the false report.

However - Remy's gossip text is still the same as Marshal Dughan's.  Minor issue, but wondering if you are able to repro?

**Comment by xfurry** [comment]
> Thanks for the prompt reply Grz3s! I rebuilt everything and reimported the latest SD2 SQL + UDB + ACID. The DK quests work now, I think something was corrupted or out of sync with the last update I did - sorry for the false report.

Great!
Then can you please close the invalid report, so we don't have to test everything again. Thanks!

**Comment by Grz3s** [comment]
hmm ..so it looks like that you doin all manualy and u missing some stuff here:
UDB 206
```sql
-- Marshal Dughan - gossip corrected
DELETE FROM gossip_menu WHERE entry = 6159;
UPDATE creature_template SET npcFlags = 3, GossipMenuId = 11611 WHERE Entry = 240;
DELETE FROM gossip_menu WHERE entry = 11611; 
INSERT INTO gossip_menu (entry, text_id, script_id, condition_id) VALUES
(11611, 16211, 0, 0);
-- missing text added
DELETE FROM npc_text WHERE ID IN (6157,16211);
INSERT INTO npc_text (ID, text0_0, text0_1, lang0, prob0, em0_0, em0_1) VALUES 
(16211, 'Ach, it\'s hard enough keeping order around here without all these new troubles popping up!  I hope you have good news, $n...', 'Ach, it\'s hard enough keeping order around here without all these new troubles popping up!  I hope you have good news, $n...', 0, 1, 5, 5),
(6157, 'Hey there, friend. My name\'s Remy. I\'m from Redridge to the east, and came here looking for business, looking for business. You got any...got any??', 'Hey there, friend. My name\'s Remy. I\'m from Redridge to the east, and came here looking for business, looking for business. You got any...got any??', 0, 1, 0, 0);
```
and this one
UDB209
```sql
-- Remy "Two Times"
UPDATE creature_template SET npcFlags = 3, GossipMenuId = 5032 WHERE entry = 241;
DELETE FROM gossip_menu WHERE entry = 5032;
INSERT INTO gossip_menu (entry, text_id, script_id, condition_id) VALUES
(5032, 6157, 0, 0);
DELETE FROM npc_text WHERE ID = 6157;
INSERT INTO npc_text (ID, text0_0, text0_1, lang0, prob0, em0_0, em0_1) VALUES 
(6157, 'Hey there, friend. My name\'s Remy. I\'m from Redridge to the east, and came here looking for business, looking for business. You got any...got any??', 'Hey there, friend. My name\'s Remy. I\'m from Redridge to the east, and came here looking for business, looking for business. You got any...got any??', 0, 1, 0, 0);
```
```code

**Comment by zimms** [comment]
I'm using the InstallFullUDB.sh script, not applying any UDB .sql files manually (though not from an empty DB, I'm using it to update using option 1).  Is that the correct way to apply updates?  I can see the strings in the npc_text DB, so they are there.

Wonder if I need to manually clean out npc_text and gossip_menu and reapply?

**Comment by Grz3s** [comment]
https://github.com/unified-db/Database/blob/master/InstallFullDB.sh
what option 1 ?
is this a same installFullDB.sh we talking about?

**Comment by zimms** [comment]
Oops sorry - ignore "option 1" - that's from when I used the batch file in a previous (unrelated) attempt at installing (which I'm now assuming isn't up-to-date).  As for InstallFullDB.sh, I only update the ACID_PATH in the config and run, and it pulls all the updates.

**Comment by Grz3s** [comment]
yes with this one ... you should see smth like this:
http://s4.postimg.org/hot7vn9wd/udb.jpg
up to 214
and im sure u wont see any problem with gossip anymore

**Comment by zimms** [comment]
I found a resolution, but it's still quite odd.

First, I traced Remy's (ID 241) gossip from creature_template (gossip ID 5032), which references npc_text ID 6157, which is your updated text in UDB209, so all should be fine, yet it wasn't...

I assume caching was going on, because even if I removed Dughan's gossip row in npc_text (and restarted mysql/mangosd/realmd), Remy would still use the text.

So to "resolve" this, I changed the gossip IDs (Npc_text ID 6157 -> 61570, Gossip_menu ID 5032 -> 50320), and Remy finally has the right text.  Therefore, I wonder if there's a collision in IDs here or something, but nothing obvious appears (after all, the IDs are unique keys), or maybe my DB is in a bad state and needs a reinstall...

Thanks for your help.  Sorry for so many questions for such a minor issue.
